### PR TITLE
feat:cxar-919 quality gates

### LINKED ISSUE #5: QG4 checks Release 3.2
## QG checks
Please keep this issue open until QG 4 is concluded and will be managed by the Issue Creator!
We will inform you about finding and proposals in separated issues, this issue here is for the Overview of the Checks!

Product Name:   
Product Owner:   
Helm Chart Version:   
App Version:   

### Check of Tractus-X Release Guidelines

This QG 4 Check is depending on the mandatory information from our current [Release Guidelines](https://eclipse-tractusx.github.io/docs/release).

#### TRG 1 Documentation
- [x] [TRG 1.01](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-1) appropriate `README.md` 
- [x] [TRG 1.02](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-2) appropriate `INSTALL.md`
- [ ] [TRG 1.03](https://eclipse-tractusx.github.io/docs/release/trg-1/trg-1-3) appropriate `CHANGELOG.md`
#### TRG 2 Git
- [x] [TRG 2.01](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-1) default branch is named `main`
- [x] [TRG 2.03](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-3) repository structure
  <details>
  <summary>Checks within TRG 2.03</summary>

  - [x] TRG 2.03 `/docs` directory contains detailed product related documentation for the Tractus-X product 
  - [x] TRG 2.03 `/charts` directory contains the Helm chart for the Tractus-X product IF available
  - [x] TRG 2.03 `AUTHORS.md` file (optional) (TRG 2.03)
  - [x] TRG 2.03 `CODE_OF_CONDUCT.md` file (TRG 2.03)
  - [x] TRG 2.03 `CONTRIBUTING.md` file (TRG 2.03)
  - [x] TRG 2.03 `DEPENDENCIES` file(s) with up to date content (Dash tool generated) (TRG 2.03)
  - [x] TRG 2.03 `LICENSE` file (TRG 2.03)
  - [x] TRG 2.03 `NOTICE.md` file (TRG 2.03)
  - [x] TRG 2.03 `SECURITY.md` file (TRG 2.03)
  </details>

- [x] [TRG 2.04](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-4) Leading product repository
  <details>
  <summary>Checks within TRG 2.04</summary>
  
  - [x] TRG 2.04 repository name must be _productname_ without prefix or suffix
  - [x] TRG 2.04 should contain the release
  - [x] TRG 2.04 references/urls to the product's other repositories
  - [x] TRG 2.04 might contain product helm chart(s)
  - [x] TRG 2.04 README.md: contains the urls for the underlying applications
  </details>

- [x] [TRG 2.05](https://eclipse-tractusx.github.io/docs/release/trg-2/trg-2-5) `.tractusx` metafile in a proper format
#### TRG 3 Kubernetes
- [x] [TRG 3.02](https://eclipse-tractusx.github.io/docs/release/trg-3/trg-3-2) PersistentVolume and PersistentVolumeClaim is used when needed
#### TRG 4 Container
- [ ] [TRG 4.01](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-1) [semantic versioning](https://semver.org/) and tagging 
- [x] [TRG 4.02](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-2) top level `README.md` file, that contains information about the used base image 
- [x] [TRG 4.03](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-3) Image has `USER` command and Non Root Container
  <details>
  <summary>Checks within TRG 4.03</summary>
  
  - [x] TRG 4.03 `deployment.yaml` has `runAsUser` and `allowPrivilegeEscalation: false` properly set
  </details>

- [x] [TRG 4.05](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05) released image must be place `DockerHub` as mandatory container registry; remove `GHCR` references 
- [x] [TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06) Notice File for `DockerHub` has all necessary information
  <details>
  <summary>Checks within TRG 4.06</summary>

  - [x] TRG 4.06 Link to the source of your base image (Container registry and GitHub if available)
  - [x] TRG 4.06 Link to your product image on `DockerHub`
  - [x] TRG 4.06 Link to your repository on `GitHub`
  - [x] TRG 4.06 Direct link to the Dockerfile used to build your image
  - [x] TRG 4.06 Link to LICENCE file in your repo as `Project License` (make clear, that this is the PROJECT licence, not an image license
  </details>

#### TRG 5 Helm
- [ ] [TRG 5.01](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-01) Helm chart must be released
  <details>
  <summary>Checks within TRG 5.01</summary>

  - [x] TRG 5.01 appropriate semantic versioning for `version` and `appVersion` has to be used in `Chart.yaml`
  - [x] TRG 5.01 must not contain any environment specific `values-xyz.yaml`
  - [x] TRG 5.01 `values.yaml` file must contain proper default values/placeholders
  - [x] TRG 5.01 No hostname provided for ingress
  - [x] TRG 5.01 Ingress is disabled
  - [x] TRG 5.01 No references to any secret engine service (e.g.: Hashicorp Vault)
  - [x] TRG 5.01 Dependencies should be prefixed with the nameOverride and/or fullnameOverride properties
  - [x] TRG 5.01 Image tag is set to the `Chart.yaml` `appVersion` property
  - [ ] TRG 5.01 must be deployable to any environment without overwriting default values with a simple helm install command
  - [x] TRG 5.01 dependencies have to be declared in Chart.yaml NOT requirements.yml
  </details>
  
- [x] [TRG 5.02](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-02) Helm chart location in `/charts` directory and correct structure
  <details>
  <summary>Checks within TRG 5.02</summary>
  
  - [x] TRG 5.02 each file must contain the [Apache 2.0 Licence](https://github.com/catenax-ng/foss-example/blob/main/general/LICENSE)
  - [x] TRG 5.02 latest tag is not used in helm chart be default

  ``` markdown
  charts/ 
      chartNameA/
        Chart.yaml
        ... 
      chartNameB/
        Chart.yaml
        ...
  AUTHORS.md 
  DEPENDENCIES.md 
  LICENCE 
  README.md 
  ```
  </details>

- [x] [TRG 5.04](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-04) CPU and memory limits and requests are properly set
- [x] [TRG 5.06](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-06) application must be configurable through the Helm chart 
- [x] [TRG 5.07](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-07) dependencies are present in the `Chart.yaml` they are properly configured
- [x] [TRG 5.08](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-08) a product has a single deployable helm chart that contains all components 
  <details>
  <summary>Checks within TRG 5.08</summary>

  - [x] TRG 5.08 name of the Chart should be just the product-name without prefix or suffix
  - [x] TRG 5.08 values file should contain all available variables (even from subcharts) with default values and comments about what they do
  - [x] TRG 5.08 helm install command should successfully install the chart to any supported Kubernetes version cluster (without overwriting default values)
  - [x] TRG 5.08 helm test runs without errors
  </details>
  
- [x] [TRG 5.09](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09) Helm Test running properly
  <details>
  <summary>Checks within TRG 5.09</summary>

  - [x] TRG 5.09 A GitHub action exist which builds or uses the helm chart which gets released
  - [x] TRG 5.09 The GitHub action can be triggered manually through Github WebUI manually running a workflow
  - [x] TRG 5.09 Helm test verifies that the application is up and running
  </details>
- [x] [TRG 5.10](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-10) Products need to support 3 versions at a time
  <details>
  <summary>Checks within TRG 5.10</summary>

  - [x] TRG 5.10 latest (K8s version 1.25)
  - [x] TRG 5.10 latest - 1 (K8s version 1.24)
  - [x] TRG 5.10 latest - 2  (K8s version 1.23)
  </details>
- [x] [TRG 5.11](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-11) Upgradeability PRERELEASE
  <details>
  <summary>Checks within TRG 5.11</summary>

  - [x] TRG 5.11 Based on the Helm test workflow, you must provide a GitHub action which takes the latest released helm chart, does an installation of it and then execute the upgrade to the current / new version.
  </details>
  
#### TRG 6 Released Helm Chart 
- [x] [TRG 6.01](https://eclipse-tractusx.github.io/docs/release/trg-6/trg-6-1) Released Helm Chart 
#### TRG 7 Open Source Governance
- [x] [TRG 7.01](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-01) Legal Documentation
- [x] [TRG 7.02](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) License and copyright header  
- [x] [TRG 7.03](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-03) IP checks for project content 
- [x] [TRG 7.04](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04) IP checks for 3rd party content
  <details>
  <summary>Checks within TRG 7.04</summary>

  - [x] TRG 7.04 DEPENDENCIES file is up-to-date and reflects the current use of the 3rd party content
  - [x] TRG 7.04 all libraries listed there should have the status "approved"
  - [x] TRG 7.04 no libraries with status "rejected"
  - [x] TRG 7.04 for libraries with status "restricted", the according IP issues must be present (issue number in the source column)
  </details>
- [ ] [TRG 7.05](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-05) Legal information for distributions
- [x] [TRG 7.06](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-06) Legal information for end user content
- [x] [TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07) Legal notice for documentation

### LINKED ISSUE #18: Publish Docker images on DockerHub
## TL;DR

- Build workflow needs fixing, so the Docker build and publish process is successfully run
- Suggestion: choose more descriptive names for Docker images
- Add "Notice for Docker Images" for each image build by this repo. Description in [TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06)
- Clarify legal info on `ontop/ontop` for ["provisioning" image](https://github.com/eclipse-tractusx/knowledge-agents/blob/main/provisioning/src/main/docker/Dockerfile)

## Description 

As described in [TRG 4.05](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-05/), all Tractus-X Docker images must be published on DockerHub. The [current workflow](https://github.com/eclipse-tractusx/knowledge-agents/blob/main/.github/workflows/build.yml), that is supposed to build and publish the images, did not have a successful run yet. 

Besides the successful run, there is a critical step missing: 
[TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06) is describing necessary legal documentation, that must be included in the repository and also used as description on the images DockerHub pages. Since the knowledge-agents repository does contain multiple Docker image, I would suggest to place dedicated Notices as .md file directly beside the Dockerfiles. This makes it easier to annotate different base images.

The [ontop/ontop](https://hub.docker.com/r/ontop/ontop) image is also not yet added to the aligned base image list of [TRG 4.02](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-02). Looking at the [GitHub repo](https://github.com/ontop/ontop), the License should be fine, but we need to collect the necessary info for the legal notice

blocks #5

### LINKED ISSUE #19: `LICENSE` missing for Helm Charts
## Description

We need to distribute Legal information with our release artifacts. For Helm Charts, we need to include the Apache-2.0 License in the packaged .tgz files.
This can be done by just placing a copy of the Apache-2.0 License in the Charts root directory (i.e. `/charts/conforming-agent/LICENSE`). 
This is also described in [TRG 5.02](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-02#the-license-file)

blocks #5

### LINKED ISSUE #20: Enhance Chart testing workflow
## TL;DR

- Add a `workflow_dispatch` trigger to the helm test workflow
- Add an input variable for [kindest/node image](https://hub.docker.com/r/kindest/node), so test can be run on different kubernetes versions 

## Description 

The current [helm-chart-lint.yaml](https://github.com/eclipse-tractusx/knowledge-agents/blob/main/.github/workflows/helm-chart-lint.yml) workflow is only triggered by automated events (PRs and pushes). 

To make quality gate reviews easier, there should also be way to trigger the run on-demand (`workflow_dispatch`). The on-demand run should test the same things, that the existing helm test workflow is already testing. Additionally, there should be a possibility to test on different kubernetes versions. This can be achieved, by specifying the [kindest/node](https://hub.docker.com/r/kindest/node/tags) image, that is used by the kind action

## Additional info

If combining `workflow_dispatch` with input variables and automated triggers does not work well in a single workflow, separate workflows could be chosen as a more suitable approach

related to #5. Checks can be done manually for this release, once the used Docker images are pushed to DockerHub

### PR DESCRIPTION
**Pull request by drcgjung**
## WHAT

- Debugged the build workflow (wrong foldernames)
- follow release guidelines including chart testing

## WHY

- Initial contribution still had TRG violations.
- Actions could not be really tested on catenax-ng fork.

## FURTHER NOTES

- The final release branch/workflows (v1.9.5) still need to be setup as main stays on -SNAPSHOT

Largely Closes #5 
Closes #20
Closes #19
Merge Hopefully Closes #18

### CODE REVIEW

**Comment by drcgjung** [comment]
@SebastianBezold I really appreciate your help, still struggling a little bit with the chart testing workflow, guess that will be resolved in a few minutes

**Comment by SebastianBezold** [comment]
Hi @drcgjung, now worries, we will get that to work :)
Unrelated to that issue, I would kindly ask you to not include the Jira links in your commit messages all the time. Reason:
- This is a closed system, that not everyone in the open-source community has access to
- Information will be lost, as soon as consortium phase is over
- The Link at the beginning of the commit message makes it hard to even see the actual useful info after the link

In best case, the Info, that you are linking to is either included in the PR description itself, or a related GitHub issue.

File: provisioning/README.md [hunk 1]
```diff
@@ -265,6 +265,24 @@ WHERE {
...
+
+Eclipse Tractus-X product(s) installed within the image:
+GitHub: https://github.com/eclipse-tractusx/knowledge-agents/tree/main/provisioning
+Project home: https://projects.eclipse.org/projects/automotive.tractusx
+Dockerfile: https://github.com/eclipse-tractusx/knowledge-agents/blob/main/provisioning/src/main/docker/Dockerfile
+Project license: Apache License, Version 2.0
+
+Used base image
+
+[ontop/ontop:5.0.2](https://hub.docker.com/r/ontop/ontop/tags?page=1&name=5.0.2)
```
Comment (SebastianBezold) [inline provisioning/README.md:280]: We should not only link the used image, but provide as much additional info on it as we can.
Examples on images that we already aligned on can be found in [TRG 4.06](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-06)

if the info is provided in later PRs, i would appreciate a GitHub issue before merging the PR, so we don't forget about that

**Review by SebastianBezold — APPROVED** [review]
LGTM! @drcgjung could you have a look at the comments regarding the additional info for your used Docker base images and decide, if you want to include it in this PR, or open issues for future addition?

File: provisioning/README.md (hunk 1 above)
Comment (drcgjung) [reply provisioning/README.md:280]: I'll try to provide that today.

**Pull request merged**